### PR TITLE
Ignore verilator errors from other files

### PR DESCRIPTION
**Pull request by jagjordi**
Solves #191

### CODE REVIEW

File: src/diagnostics.rs [hunk 1]
```diff
@@ -239,10 +239,12 @@ fn verilator_syntax(
...
     static RE: std::sync::OnceLock<Regex> = std::sync::OnceLock::new();
     let re = RE.get_or_init(|| {
-        Regex::new(
-            r"%(?P<severity>Error|Warning)(-(?P<warning_type>[A-Z0-9_]+))?: [^:]+:(?P<line>\d+):((?P<col>\d+):)? ?(?P<message>.*)",
-        )
-        .unwrap()
+        let file_path_str = regex::escape(file_path.to_str().unwrap_or(""));
+        let pattern = format!(
+            r"%(?P<severity>Error|Warning)(-(?P<warning_type>[A-Z0-9_]+))?: {}:(?P<line>\d+):((?P<col>\d+):)? ?(?P<message>.*)",
+            file_path_str
```
Comment (vivekmalneedi) [inline src/diagnostics.rs:245]: The regex is created inside a oncelock to ensure that we only compile it once. So this change would bake in the file path of the first file that was checked by verilator.

**Comment by vivekmalneedi** [comment]
Thanks for filing the issue and creating this PR! I modified the regex to add a capture group for file path that we can check against. Please confirm if the fix works on your end if possible.

File: src/diagnostics.rs (hunk 1 above)
Comment (jagjordi) [reply src/diagnostics.rs]: That explains why after these changes the diagnostics only worked on the first file I opened and not the follow-up ones :) But I was not able to find out why given that as I said I am not familiar with rust :)

**Comment by hakan-demirli** [comment]
@vivekmalneedi I have tested the PR and It is working as intended. Errors from submodules are not creeping out to the top.

**Comment by vivekmalneedi** [comment]
@hakan-demirli  thanks for testing, I forgot this was still open.

**Pull request merged**